bcgov/biohubbc
# BHBC-2215, 2216: Apply 'Viewer' and 'Editor' Permissions to Project / Survey Views

# Overview

As a Project Lead, I want to make sure that people I’ve invited to my team can only see the functions their role allows them to so they aren’t getting confused as to why some functionality appears broken.

## Links to Jira tickets

 - https://quartech.atlassian.net/browse/BHBC-2215
 - https://quartech.atlassian.net/browse/BHBC-2216

## Description of relevant changes

 **- TBD.**

## Testing Procedures

0. Create a project and survey.
1. Add a user to the a project as a ‘Viewer’. The easiest way to do this would be to create a BCeID Basic account in dev. You can do this here: https://www.development.bceid.ca/register/.
2. View the project as the 'Viewer' user. It should:

 - [ ] NOT show the following interactive components:
    - Project Settings
    - Add Survey
    - Upload Documents
    - Delete Document (see individual document row items)
    - Edit Location

3. Repeat _Step 1_ and _Step 2_, but for a Survey. It should:

 - [ ] NOT show the following interactive components:
    - Survey Settings
    - Import Observations
    - Delete Observations
    - Import Summary Results
    - Delete Summary Results
    - Upload Documents
    - Delete Document (see individual document row items)
    - Edit Study Area

4. Add a user to the a project as an ‘Editor’.
5. View the project as the 'Editor' user. It should:

 - [ ] NOT show the following interactive components:
    - Delete Document (see individual document row items)
